hummingbot/connector/derivative/coinflex_perpetual/coinflex_perpetual_derivative.py

```diff
@@ -0,0 +1,1391 @@
+                if not fee_collected and len(order_update.get("fees", {})):
+                    fee_collected = True
+                    fee_data = order_update.get("fees")
+                    fee_asset = list(fee_data.keys())[0]
+                    fee_amount = decimal_val_or_none(fee_data[fee_asset])
+                else:
+                    fee_asset = tracked_order.quote_asset
+                    fee_amount = None
+                position_side = order_update.get("side", "BUY")
+                position_action = (PositionAction.OPEN
+                                   if (tracked_order.trade_type is TradeType.BUY and position_side == "BUY"
+                                       or tracked_order.trade_type is TradeType.SELL and position_side == "SELL")
+                                   else PositionAction.CLOSE)
+                flat_fees = [] if not fee_amount or fee_amount == s_decimal_0 else [TokenAmount(amount=fee_amount, token=fee_asset)]
+
+                fee = TradeFeeBase.new_perpetual_fee(
+                    fee_schema=self.trade_fee_schema(),
+                    position_action=position_action,
+                    percent_token=fee_asset,
+                    flat_fees=flat_fees,
+                )
+                trade_update = TradeUpdate(
+                    trading_pair=tracked_order.trading_pair,
+                    trade_id=int(trade_id),
+                    client_order_id=tracked_order.client_order_id,
+                    exchange_order_id=str(order_update["orderId"]),
+                    fill_timestamp=int(trade_data["timestamp"]) * 1e-3,
+                    fill_price=fill_price,
+                    fill_base_amount=exec_amt_base,
+                    fill_quote_amount=exec_amt_quote,
+                    fee=fee,
+                )
+                self._client_order_tracker.process_trade_update(trade_update=trade_update)
+
+    def _process_order_not_found(self,
+                                 client_order_id: str,
+                                 tracked_order: InFlightOrder) -> bool:
+        self._order_not_found_records[client_order_id] = (
+            self._order_not_found_records.get(client_order_id, 0) + 1)
+        if (self._order_not_found_records[client_order_id] >=
+                self.MAX_ORDER_UPDATE_RETRIEVAL_RETRIES_WITH_FAILURES):
+            # Wait until the order not found error have repeated a few times before actually treating
+            # it as failed. See: https://github.com/CoinAlpha/hummingbot/issues/601
+
+            order_update: OrderUpdate = OrderUpdate(
+                client_order_id=client_order_id,
+                trading_pair=tracked_order.trading_pair,
+                update_timestamp=self.current_timestamp if self.current_timestamp != NaN else int(time.time()),
+                new_state=OrderState.FAILED,
+            )
+            self._client_order_tracker.process_order_update(order_update)
+            return True
+        return False
```

Comment:
The order not found should be processed by using the method `process_order_not_found` in the client order tracker.

Reply:
Unfortunately the `process_order_not_found` method in the client order tracker does not return booleans as in my method.

Reply:
Do we need to raise the exception in the cancel method if the order was not found but we have not reached the limit of max not found times? I think the cancel logic will work correctly without that exception being raised. And the other place where the `process_order_not_found` is called is not using the boolean result. I still believe it should be possible to use the logic in the order book tracker to handle the situation.